flink-connector-oceanbase-base/src/main/java/com/oceanbase/connector/flink/table/DataChangeRecord.java

```diff
@@ -36,11 +37,15 @@ static KeyExtractor simple() {
             return record ->
                     Optional.ofNullable(record.getTable().getKey())
                             .map(
-                                    keys ->
-                                            new DataChangeRecordData(
-                                                    keys.stream()
-                                                            .map(record::getFieldValue)
-                                                            .toArray()))
+                                    keys -> {
+                                        Object[] array =
+                                                keys.stream().map(record::getFieldValue).toArray();
+                                        if (array.length == 0) {
```

Comment:
What's it for? The process should directly go to `orElse` if the record is from a non-pk table,

Reply:
https://github.com/oceanbase/flink-connector-oceanbase/blob/f7f175f62991c63fb63a1e30c9a4ca0af60420ee/flink-connector-oceanbase-base/src/main/java/com/oceanbase/connector/flink/sink/OceanBaseWriter.java#L132-L144

- If record.getTable().getKey() is an empty but not null primary key List, this `KeyExtractor.simple()` will return an empty DataChangeRecord object but not NULL
- Then line 133 of the code block above will return false, causing the non-primary key table to be mistakenly treated as the primary key table.

```
        static KeyExtractor simple() {
            return record ->
                    // when record.getTable().getKey() is empty List
                    Optional.ofNullable(record.getTable().getKey())
                            .map(
                                    keys ->
                                            new DataChangeRecordData(
                                                    keys.stream()
                                                            .map(record::getFieldValue)
                                                            .toArray()))
                            .orElse(null);
        }
```